web/pdf_viewer.js

```diff
@@ -2098,15 +2123,19 @@ class PDFViewer {
     this.#annotationEditorUIManager.updateParams(type, value);
   }
 
-  refresh() {
+  refresh(update, updateArgs = Object.create(null)) {
     if (!this.pdfDocument) {
       return;
     }
-    const updateArgs = {};
     for (const pageView of this._pages) {
       pageView.update(updateArgs);
     }
-    this.update();
+    if (this.#scaleTimeoutId !== null) {
+      clearTimeout(this.#scaleTimeoutId);
```

Comment:
Let's null it as well, to prevent this code-path from running more than once.
```suggestion
      clearTimeout(this.#scaleTimeoutId);
      this.#scaleTimeoutId = null;
```